### DISCUSSION TITLE
How to call all resources of a computer for calculation

### DISCUSSION THREAD

**Discussion by 0batsy0** [phoebe-project/phoebe2#721] [General]
When I run ‘run_solver’，it will consume a lot of time.So I want to use all the resources of my computer to speed up computation,What shall I do?

**Comment by kecnry** [comment]
Running MCMC with a full-fledged phoebe model can definitely be computationally expensive, but there are a few different things you can do to [speed up phoebe computations](http://phoebe-project.org/docs/latest/tutorials/optimizing) depending on your setup, you can make use of [MPI](http://phoebe-project.org/docs/latest/tutorials/mpi) or [multiprocessing](http://phoebe-project.org/docs/latest/tutorials/multiprocessing), or you can run the `run_solver` job on [external compute resources](http://phoebe-project.org/docs/latest/tutorials/server).

**Reply by 0batsy0** [reply]
![cpu](https://user-images.githubusercontent.com/130527283/232649442-65a46085-0bd4-479d-9318-95178e24149c.PNG)
As shown in the picture, I am certain that I have set up all multiprocessing, but I still haven't used all the CPU resources

**Reply by kecnry** [reply]
MPI in PHOEBE will intelligently switch between parallelizing per-time in the forward model or per-walker in MCMC depending on the number of walkers (if `nprocs <= nwalkers` then it will parallelize per-walker, otherwise it'll parallelize per-time-step that requires a mesh to be built or populated).  Multiprocessing, I believe, only supports per-walker parallelizing, so if you're only using 16 walkers, then it will likely only use up to 16 processors.  If you want to make the most out of 80 processors, I'd suggest either using 80 walkers or using MPI and try to set your [compute/solver times](http://phoebe-project.org/docs/latest/tutorials/solver_times) to some integer multiple of 80.
